Makefile

```diff
@@ -84,6 +84,13 @@ endif
 
 GO_TEST_GINKGO_ARGS ?= -test.v -ginkgo.v -ginkgo.failFast
 
+ifeq ($(origin DOCKERCMD),undefined)
+DOCKERCMD?=$(shell docker version >/dev/null 2>&1 && echo docker)
+ifeq ($(DOCKERCMD),)
+DOCKERCMD=$(shell podman version >/dev/null 2>&1 && echo podman)
+endif
+endif
```

Comment:
Podman is available everywhere, and is a drop-in replacement for docker, so we can
safely use podman instead of docker and document that we require it.

If we want to allow using docker instead of podman we can provide a environment
variable like CONTAINER_TOOL and let the user override it without any checks.

If the tool is not available the command will fail in a pretty clear way:

    $ docker build -t foo .
    bash: docker: command not found...

Maintaining cryptic makefiles is not fun, better to keep it simple.

Reply:
Being able to use `alias docker=podman` is nice, but it is a hack and I don't want hacks
in my environment. I want to know exactly what is running and be able to run both podman
and docker if needed. Installing podman-docker is a better hack also handling man pages
and doing system level change cleanly.

But do have an issue switching to podman? This should affect only ramen developers.

Reply:
I really do not have an issue, it's just that operator sdk devs (who visit/contribute in the future) would have something new to deal with, which we can avoid. I have seen this DOCKERCMD or variants in a few Makefiles on other projects, so not a blocker from my end, just stating there are alternatives out there.

Reply:
Looking at this a bit more, "undockerizing" this for the env variable is only a start, as we have various docker* makefile targets as well. As a result we can just retain a simple environment variable (say DOCKERCMD) with it's default as `podman` and let users override this in their shell as required. (As @nirs states keeps the Makefile a little simpler)

Reply:
Updated the patch to default to podman.